### PR TITLE
[New KB Article] Semgrep scan troubleshooting (initial pass)

### PR DESCRIPTION
**Pull request by armchairlinguist**
We currently have some troubleshooting content, but not a general guide to the common troubleshooting steps we recommend.

Most likely we will want to reorganize and enhance this information over time, as well as connect it to specialized articles for further troubleshooting, but I wanted to get started with an initial pass for now.

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team

### CODE REVIEW

File: docs/kb/semgrep-code/semgrep-scan-troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+description: Troubleshoot common issues with Semgrep scans.
+---
+
+# A Semgrep scan is having a problem - what next?
+
+If a Semgrep scan is failing or running slowly, try the following steps to investigate:
+
+1. [Update Semgrep](/docs/upgrading/) to the latest version, if you are not currently running the latest version. Some errors result from an older version of Semgrep being used with newer rules.
+2. Re-run the scan with either the `-v`/`--verbose` or `--debug` (extremely verbose) flags. These options provide more information about what is failing.
+3. Running Semgrep Pro Engine in the scan? Remove any options starting with `--pro`, or run Semgrep with `--oss-only`. This allows isolation of any issues related to Semgrep Pro Engine, and often speeds up a scan or reduces memory usage.
```
Comment (s-santillan) [inline docs/kb/semgrep-code/semgrep-scan-troubleshooting.md:14]:
```suggestion
3. If you are running Semgrep Pro Engine in the scan, remove any options starting with `--pro`, or run Semgrep with `--oss-only`. This allows isolation of any issues related to Semgrep Pro Engine, and often speeds up a scan or reduces memory usage.
```
Just concerned that non-native English readers will be confused by the gerund ending in a question.

**Review by s-santillan — APPROVED** [review]
lgtm! very useful tips.

**Comment by armchairlinguist** [comment]
I had this as a draft in case it might need a major re-write, but Sara likes it, and I feel pretty happy with it as a first pass, so marking it ready for actual review. :)

File: docs/kb/semgrep-code/semgrep-scan-troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+If you receive a 401 when scanning using registry rules (for example, with `--config auto`), try the following:
+
+1. Run `semgrep logout`.
+2. Attempt the scan again.
+
+If this is successful, your stored local token was invalid. Use `semgrep login` to log in again and receive a fresh token.
+
+## Scan failures with analysis errors
+
+Analysis or parsing errors usually only affect a particular rule, file, or language. If your scan encounters an analysis issue, using verbose logging can provide you with helpful error details, such as `metavariable-pattern failed because we lack range info for $X, please file a bug report`.
```
Comment (sinat101) [inline docs/kb/semgrep-code/semgrep-scan-troubleshooting.md:43]: since this is a full on sentence would putting it in a code block be better?

**Review by sinat101 — CHANGES_REQUESTED** [review]

File: docs/kb/semgrep-code/semgrep-scan-troubleshooting.md (hunk 2 above)
Comment (armchairlinguist) [reply docs/kb/semgrep-code/semgrep-scan-troubleshooting.md:43]: very possible! let me test that locally

Comment (sinat101) [reply docs/kb/semgrep-code/semgrep-scan-troubleshooting.md:43]: Cool, this is a minor thing so I am just going to approve, thanks :D

**Review by sinat101 — APPROVED** [review]

**Review by sinat101 — APPROVED** [review]

**Pull request merged**